helpers/controllers/passwordController.ts

```diff
@@ -25,21 +24,29 @@ export const reqPwReset = async (
       throw new UserInputError('Please provide username or email')
     }
 
-    const user = await findUser(userOrEmail)
+    let user = await findUser(userOrEmail)
 
     if (!user) {
       throw new UserInputError('User does not exist')
     }
 
-    const encodedToken = encode({
+    const forgotToken = encode({
       userId: user.id,
       userToken: nanoid()
     })
+    const tokenExpiration = new Date(Date.now() + THREE_DAYS)
 
-    user.forgotToken = encodedToken
-    user.tokenExpiration = new Date(Date.now() + THREE_DAYS)
-    await user.save()
-    sendResetEmail(user.email, user.forgotToken)
+    user = await prisma.user.update({
+      where: {
+        id: user.id
+      },
+      data: {
+        forgotToken,
+        tokenExpiration
+      }
+    })
+
+    sendResetEmail(user.email!, user.forgotToken!)
```

Comment:
Why `user.email` is non-nullable in this case? `user` is just a record of `User` type and according to prisma schema email can be null.

Reply:
Because we can be sure the user has an email if they're changing their password, since they receive an email for that. I didn't want to add useless checks just to keep TS happy.

I don't like this either, but the migration to make email and username not null is a bit harder. There's some entries with null or duplicated values in the database and I'm not sure how we should handle data loss in this case.

Reply:
Makes sense. I wonder how many users don't have email. Probably very few and they were created long ago, so we maybe we can simply delete them for inactivity.

Reply:
I think we should delete them if they're inactive and make `email` non-nullable in prisma schema. 

But we will need a way to confirm their last activity.